apps/extension/src/shared/components/default-frontend-form/index.tsx

```diff
@@ -1,39 +1,39 @@
+import { useMemo, useRef } from 'react';
+import type { EntityMetadata } from '@penumbra-labs/registry';
 import { SelectList } from '@repo/ui/components/ui/select';
+import { Button } from '@repo/ui/components/ui/button';
 import { AllSlices } from '../../../state';
 import { useStoreShallow } from '../../../utils/use-store-shallow';
-import { useMemo, useRef } from 'react';
-import { Button } from '@repo/ui/components/ui/button';
 import { NewFrontendInput } from './new-frontend-input';
 import { useIsFocus } from './use-is-focus';
-import { extractDomain } from './extract-domain';
 import { LoadingList } from '../loading-list';
 import { useRegistry } from '../registry';
 
-interface DisplayedFrontend {
-  title: string;
-  url: string;
-}
-
 const useFrontendsList = (selectedRpc?: string) => {
   const { data, isLoading, error } = useRegistry();
 
-  const frontends = (data?.frontends ?? []).map(frontend => ({
-    title: extractDomain(frontend),
-    url: frontend,
-  }));
+  const frontends = useMemo(() => {
+    const arr: EntityMetadata[] = data?.frontends ?? [];
+
+    if (selectedRpc) {
+      return [
+        ...arr,
+        {
+          name: 'Embedded RPC frontend',
+          /*NB: we merge using the variadic URL constructor here to avoid double-slashes*/
```

Comment:
question: what's `NB` mean? 🤔

Reply:
no idea, this comment was there before me 🙈